### PR TITLE
Datetime error message (unable to continue that PR)

### LINKED ISSUE #10720: DOC/ERR: better error message on unsuccessful datetime parsing
exceptions that are raised on unsuccessful datetime/timedelta parsing should add this:

``you can coerce to NaT by passing errors='coerce'``

comment at the end: https://github.com/pydata/pandas/pull/10674

### PR DESCRIPTION
**Pull request by baevpetr**
- [ ] closes #10720
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/core/arrays/datetimes.py [hunk 1]
```diff
@@ -1970,6 +1970,15 @@ def objects_to_datetime64ns(
             #  return them as i8 to distinguish from wall times
             return values.view("i8"), tz_parsed
         except (ValueError, TypeError):
+            # GH#10720. If we failed to parse datetime then notify
+            #  that flag errors='coerce' could be used to NaT.
+            #  Trying to distinguish exception based on message.
+            if "Unknown string format" in e.args[0]:
```
Comment (WillAyd) [inline pandas/core/arrays/datetimes.py]: Can you use `raise from` instead? I think makes things clearer with Python3

File: pandas/core/arrays/datetimes.py [hunk 2]
```diff
@@ -1970,6 +1970,15 @@ def objects_to_datetime64ns(
...
+                msg = (
+                    " ".join(e.args)
+                    + ". You can coerce to NaT by passing errors='coerce'"
```
Comment (WillAyd) [inline pandas/core/arrays/datetimes.py]: Let's opt for f-strings over adding them like this

File: pandas/tests/indexes/datetimes/test_tools.py [hunk 3]
```diff
@@ -480,6 +480,17 @@ def test_to_datetime_unparseable_ignore(self):
         s = "Month 1, 1999"
         assert pd.to_datetime(s, errors="ignore") == s
 
+    def test_to_datetime_unparseable_raise(self):
+        # GH#10720
+        invalid_data = "Month 1, 1999"
+        expected_args = (
+            f"Unknown string format: {invalid_data}. "
+            f"You can coerce to NaT by passing errors='coerce'"
```
Comment (WillAyd) [inline pandas/tests/indexes/datetimes/test_tools.py]:
```suggestion
            "You can coerce to NaT by passing errors='coerce'"
```

Don't need the f prefix if not parametrizing anything

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Comment by baevpetr** [comment]
Tests passed locally, but failed in Ci.
On some platforms
1) There's extra **%s** flag:
`AssertionError: Pattern "Unknown string format: Month 1, 1999. You can coerce to NaT by passing errors='coerce'" not found in "Unknown string format: %s Month 1, 1999. You can coerce to NaT by passing errors='coerce'"
`
2) or date is missing at all:
`AssertionError: Pattern "Unknown string format: Month 1, 1999. You can coerce to NaT by passing errors='coerce'" not found in "Unknown string format. You can coerce to NaT by passing errors='coerce'"`

File: pandas/core/arrays/datetimes.py [hunk 4]
```diff
@@ -1959,6 +1959,13 @@ def objects_to_datetime64ns(
             #  return them as i8 to distinguish from wall times
             return values.view("i8"), tz_parsed
         except (ValueError, TypeError):
+            # GH#10720. If we failed to parse datetime then notify
+            #  that flag errors='coerce' could be used to NaT.
+            #  Trying to distinguish exception based on message.
+            msg = " ".join(e.args)
```
Comment (WillAyd) [inline pandas/core/arrays/datetimes.py]:
```suggestion
            msg = str(e)
```

**Review by WillAyd — CHANGES_REQUESTED** [review]
Might help with the CI failure as well

**Comment by baevpetr** [comment]
`str(e)` doesn't pass even locally. Output looks like: 
`"('Unknown string format:', 'Month 1, 1999'). You can coerce to NaT by passing errors='coerce'"`
Tried to use list comps. Doesn't help too.

**Comment by baevpetr** [comment]
ping @WillAyd

**Comment by baevpetr** [comment]
ping №2: @jbrockmendel

File: pandas/core/tools/timedeltas.py [hunk 5]
```diff
@@ -113,9 +113,10 @@ def _coerce_scalar_to_timedelta_type(r, unit="ns", errors="raise"):
 
     try:
         result = Timedelta(r, unit)
-    except ValueError:
+    except ValueError as e:
```
Comment (jbrockmendel) [inline pandas/core/tools/timedeltas.py]: can you call this "err" instead of "e".  i really dislike 1-letter variable names

Comment (baevpetr) [reply pandas/core/tools/timedeltas.py]: even here. ok!)

**Pull request closed without merging**